src/v2/BaseLoginRouter.js

```diff
@@ -125,7 +126,7 @@ export default Router.extend({
     // transform response
     const ionResponse = transformIdxResponse(this.settings, idxResponse, lastResponse);
 
-    this.appState.setIonResponse(ionResponse);
+    await this.appState.setIonResponse(ionResponse, this.hooks);
```

Comment:
It would be much cleaner if somehow hooks can be available under `ionResponse` object.. possible if we can pass this in `transformIdxResponse` instead?..

Reply:
why would it be cleaner if the hooks are part of the ionResponse? They are not related to the response, they are a configured part of the client state.

Reply:
if anything, they could be part of appState, but again because they need to be configured before or after render (when app state is created), I put them in their own model.

Reply:
I wanted to have `hooks` as part settings or options or something instead of standalone.. But it seems it's not possible because of how we want to execute and assign them..

Reply:
It could be done with a bigger refactor, which I considered and even started, but I decided I did not want to mess with how Settings and AppState are constructed with this PR, to avoid possibility of creating regressions. If we constructed either of these objects before render they could be a good home for this feature. Settings was my first choice, AppState could also make sense (although it gets cleared regularly).  The main reason Settings is created late is because options from the constructor are merged with options passed to renderEl - a refactor would have to handle merging in renderEl options late. I'm not sure we have sufficient unit tests to be confident in making this change now. I can create a new ticket for a Settings refactor to include hooks, renderEl options, and unit tests.

Reply:
OKTA-432376